File: macos/spotter-macOS/Clipboard/Clipboard.m
#import "React/RCTBridgeModule.h"
#import "React/RCTEventEmitter.h"

@interface RCT_EXTERN_MODULE(NativeClipboard, RCTEventEmitter)

//RCT_EXTERN_METHOD(getValue resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(getValue: (RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(setValue: (NSString)value)

@end

File: macos/spotter-macOS/Clipboard/Clipboard.swift
import Foundation
import Magnet
import Carbon

@objc(NativeClipboard)
class NativeClipboard: RCTEventEmitter {

  @objc
  func setValue(_ value: NSString) -> Void {
    NSPasteboard.general.clearContents()
    NSPasteboard.general.setString(value as String, forType: .string)
  }
  
  @objc
  func getValue(_ resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
    resolve(NSPasteboard.general.string(forType: .string))
  }
  
  override func supportedEvents() -> [String]! {
    return []
  }
  
  @objc override static func requiresMainQueueSetup() -> Bool {
    return false
  }
  
}
